tools/freeze_requirements.py

```diff
@@ -0,0 +1,75 @@
+import os
+from concurrent.futures import ThreadPoolExecutor
+from subprocess import run
+from typing import Tuple
+
+from tools.lib import gitroot
+from tools.lib import ts_print as print
+
+
+def worker(cmd: Tuple[str, ...]) -> Tuple[str, int]:
+    cmd_pretty = " ".join(cmd)
+    print(f"+ {cmd_pretty}")
```

Comment:
`print(f"+ {shlex.join(cmd)}")`  (I'd add as suggestion but it needs an import!)

Reply:
Ah yes, love to be on Python 3.8 😎